### PR TITLE
added testdata for crop layer #24384

### PR DESCRIPTION
**Pull request by Dhanwanth1803**
Hi @dkurt . Like you mentioned in https://github.com/opencv/opencv/pull/24384 I have added the test data files for crop layer in `darknet` folder using `/opencv_extra/testdata/dnn/darknet/generate_darknet_models.py`. Also can you tell me what additions changes that I have to do in the opencv repository to put up a pr there aswell.

### CODE REVIEW

File: testdata/dnn/darknet/avgpool_softmax_in.npy
Comment (dkurt) [inline testdata/dnn/darknet/avgpool_softmax_in.npy:1]: remove unrelated files

File: testdata/dnn/darknet/crop.cfg [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# batch=128
+# subdivisions=4
+
+# Testing
+batch=1
+subdivisions=1
+
+height=256
+width=256
+channels=3
```
Comment (dkurt) [inline testdata/dnn/darknet/crop.cfg:12]: Please choose smaller input resolution I think the following is enough for the test:
```
height=5
width=6
channels=3
```
```
crop_height=3
crop_width=2
```

File: testdata/dnn/darknet/generate_darknet_models.py [hunk 2]
```diff
@@ -37,4 +35,4 @@ def genTestData(name, inpShape, outShape):
 genTestData('route_multi', [1, 6, 3, 6], [1, 4, 2, 3])
 genTestData('relu', [1, 2, 3, 4], [1, 2, 3, 4])
 genTestData('sam', [1, 2, 3, 4], [1, 4, 3, 4])
-
+genTestData('crop', [1, 2, 3, 4], [1, 2, 3, 4])
```
Comment (dkurt) [inline testdata/dnn/darknet/generate_darknet_models.py:38]: Should shapes match the values from the .cfg file?

File: testdata/dnn/darknet/convolutional_out.npy
Comment (dkurt) [inline testdata/dnn/darknet/convolutional_out.npy:1]: Should be restored

File: testdata/dnn/darknet/generate_darknet_models.py (hunk 2 above)
Comment (Dhanwanth1803) [reply testdata/dnn/darknet/generate_darknet_models.py:38]: > Should shapes match the values from the .cfg file?

Yes. the `inpShape` should match the values from `.cfg file`. Also can you suggest me what values for `outShape` can I pass as the parameter.

Comment (dkurt) [reply testdata/dnn/darknet/generate_darknet_models.py]: I think output shape should match `crop_height` and `crop_width`. So `genTestData('crop', [1, 3, 5, 6], [1, 3, 3, 3])`

File: testdata/dnn/darknet/generate_darknet_models.py [hunk 3]
```diff
@@ -37,4 +37,5 @@ def genTestData(name, inpShape, outShape):
 genTestData('route_multi', [1, 6, 3, 6], [1, 4, 2, 3])
 genTestData('relu', [1, 2, 3, 4], [1, 2, 3, 4])
 genTestData('sam', [1, 2, 3, 4], [1, 4, 3, 4])
+genTestData('crop', [1, 5, 6, 3], [1, 3, 2, 1])
```
[the remark this answers is not in the document]
Comment (Dhanwanth1803) [reply testdata/dnn/darknet/generate_darknet_models.py:40]: Yeah okay. I will change it right away

**Comment by Dhanwanth1803** [comment]
@dkurt  I am so sorry I am new to github. I dont know how this got closed. Can i make a new PR

**Comment by dkurt** [comment]
@Dhanwanth1803, you probalby pushed a branch with no changes and GitHub closed it automatically.
